homeassistant/components/wsdot/sensor.py

```diff
@@ -44,22 +51,48 @@
 async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Migrate a platform-style wsdot to entry-style."""
+    entries = hass.config_entries.async_entries(DOMAIN)
+    if entries:
+        for entry in entries:
+            _LOGGER.info(
+                "Found already-setup WSDOT entry. Skipping platform setup. Your "
+                'configuration.yaml might contain a "wsdot" entry in `sensor.platform` '
+                "that is no longer needed"
+            )
+            if entry.data[CONF_API_KEY] != config[CONF_API_KEY]:
+                _LOGGER.warning(
+                    "Legacy Platform WSDOT entry found but there already exists a WSDOT "
+                    "entry with a different API Key. Skipping migration of this configuration"
+                )
+        return
+
+    await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_IMPORT},
+        data=config,
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Set up the WSDOT sensor."""
-    sensors = []
     session = async_get_clientsession(hass)
-    api_key = config[CONF_API_KEY]
-    wsdot_travel = WsdotTravelTimes(api_key=api_key, session=session)
-    for travel_time in config[CONF_TRAVEL_TIMES]:
-        name = travel_time.get(CONF_NAME) or travel_time.get(CONF_ID)
-        travel_time_id = int(travel_time[CONF_ID])
-        sensors.append(
-            WashingtonStateTravelTimeSensor(name, wsdot_travel, travel_time_id)
+    api_key = entry.data[CONF_API_KEY]
+    wsdot_travel_times = wsdot_api.WsdotTravelTimes(api_key=api_key, session=session)
+    for subentry_id, subentry in entry.subentries.items():
+        name = subentry.data[CONF_NAME]
+        travel_time_id = int(subentry.data[CONF_ID])
```

Comment:
why didn't we store this as an int in the first place?

Reply:
The current documentation shows an example of `id` being a yaml int, but notates that the field is a required string. So there seems to be some confusion. With the way the id was used earlier, it didn't matter which the user wrote.

I suppose with the migration I can for all platform config ids to an int.